### ISSUE TITLE
Prebuilds in npm package cause errors publishing to homebrew

### ISSUE THREAD

**Issue by jsjoeio** [parcel-bundler/watcher#93]
## Problem

`@parcel/watcher` includes all prebuilds (https://github.com/parcel-bundler/watcher/pull/34) regardless of the architecture. This breaks `brew audit` for any packages that publish to homebrew and rely on `@parcel/watcher` in their dependency tree.

### Example

[Example failure](https://github.com/Homebrew/homebrew-core/runs/4741363583?check_suite_focus=true#step:6:33):
```ruby
==> brew audit code-server --online --git --skip-style
==> FAILED
Full audit code-server --online --git --skip-style output
  code-server:
    * Binaries built for a non-native architecture were installed into code-server's prefix.
      The offending files are:
        /usr/local/Cellar/code-server/4.0.1/libexec/vendor/modules/code-oss-dev/node_modules/@parcel/watcher/prebuilds/darwin-arm64/node.napi.glibc.node	(arm64)
```

## Additional Context

We (coder/code-server) publish code-server (VS Code in the browser) to homebrew. In [October 2021](https://github.com/microsoft/vscode/issues/132483), VS Code switched to using `@parcel/watcher` as their main file watcher. Now, we're running into issues publishing code-server to homebrew.

### Workaround

Our current solution is to check the architecture and delete non-native binaries in `@parce/watcher/prebuilds`. See [example](https://github.com/Homebrew/homebrew-core/blob/c8a416bba8049ae8c5532991a7f0103b8695a565/Formula/code-server.rb#L32-L35):

```ruby
  def install
    node = Formula["node@14"]
    system "yarn", "--production", "--frozen-lockfile"
    # @parcel/watcher bundles all binaries for other platforms & architectures
    # This deletes the non-matching architecture otherwise brew audit will complain.
    prebuilds = buildpath/"vendor/modules/code-oss-dev/node_modules/@parcel/watcher/prebuilds"
    (prebuilds/"darwin-x64").rmtree if Hardware::CPU.arm?
    libexec.install Dir["*"]
    env = { PATH: "#{node.opt_bin}:$PATH" }
    (bin/"code-server").write_env_script "#{libexec}/out/node/entry.js", env
  end
  ```

**Comment by mischnic** [comment]
What do you propose we should do? 

Also, doesn't homebrew-core usually have the rather strict requirement that everything is compiled by Homebrew?

**Comment by jsjoeio** [comment]
> Also, doesn't homebrew-core usually have the rather strict requirement that everything is compiled by Homebrew?

Yes, I think you're right.

> What do you propose we should do?

Hmm...I don't know. Hard to tell if we're the only ones experiencing this. But I guess if someone else uses `@parcel/watcher` and publishes on Homebrew, they're going to run into the same issue.

If the prebuilds weren't included in the npm package, then you'd have to run some install/postinstall script to build them, but then that reopens (https://github.com/parcel-bundler/watcher/issues/30) and probably (https://github.com/parcel-bundler/watcher/issues/26).

One solution that comes to mind is publishing two different versions (one that has the prebuilds and one that builds on install based on arch) but that sounds like a maintenance cost for y'all.

Do any solutions come to mind for you?

**Comment by devongovett** [comment]
This should be solved by #144 which splits the prebuilds into their own separate packages per platform/architecture and uses optional dependencies to install them. So on macOS, you should only get darwin binaries.
